src/sonic-config-engine/minigraph.py

```diff
@@ -411,13 +411,14 @@ def parse_xml(filename, platform=None, port_config_file=None):
         elif child.tag == str(QName(ns, "DeviceInfos")):
             (port_speeds_default, port_descriptions) = parse_deviceinfo(child, hwsku)
 
-    results = {}
+    current_device = [devices[key] for key in devices if key.lower() == hostname.lower()][0]
```

Comment:
so we are assuming that there will be one and only one device that has the same hostname as the device itself's hostname?

Reply:
Yes.